### ISSUE TITLE
Solidus char customization would be nice apparently / is hardcoded

### ISSUE THREAD

**Issue by skarzhevskyy** [MenoData/Time4J#359]
example 

    ChronoFormatter<PlainTime> formatter = ChronoFormatter.setUp(PlainTime.class, Locale.US)//
            .addIgnorableWhitespace()//
            .addPattern("HH:mm", PatternType.CLDR_24)//
            .addIgnorableWhitespace()//
            .build();

  System.out.println(ClockInterval.parse("11:10 - 23:50", formatter));

**Comment by MenoData** [comment]
Good idea. Two principle ways are to be considered for the implementation (maybe even both). The actual hard-coded implementation was just motivated by ISO-standard.

- Add an extra controllable attribute for the separation char in `IsoInterval`
- Use CLDR-resources for localized separation char

**Comment by MenoData** [comment]
I have now studied CLDR and ICU4J:

http://cldr.unicode.org/translation/date-time-patterns
http://icu-project.org/apiref/icu4j/com/ibm/icu/text/DateIntervalFormat.html

It appears that a method with a signature like:

```public static ClockInterval parse(String text, ChronoParser<PlainTime> parser, String intervalPattern)```

will be sufficient and flexible enough for solving this issue (using a pattern like "{0} - {1}"). If  the text to be parsed contains a variable count of white space characters (see your usage of builder-method `addIgnorableWhitespace()` then the interval pattern "{0}-{1}" (without spaces) can be used instead.

Other thoughts:

- The existing `parse()`-methods without the `intervalPattern`-parameter would try to use the `intervalFormatFallback`-entries in CLDR-data but for `Locale.ROOT` still use the ISO-slash. The locale can be evaluated by determining if given parser is of type `ChronoFormatter` (if yes then cast and ask for the locale).

- A similar approach can be chosen for `DateInterval`, `TimestampInterval` and `MomentInterval`.

- The `print()`-methods in class `IsoInterval` can be overloaded by using the extra `intervalPattern`-parameter. The existing methods can try to get the interval pattern via the locale and fall back to the ISO-slash if the pattern is not available.

- For date intervals, there are more possibilities to consider which I would postpone to the future (due to its internal complexity - namely support for abbreviated formats like "d – d MMM y").

**Comment by MenoData** [comment]
Now I have released v3.9 and v4.6. You can solve your problem in various ways. Look at following JUnit-tests:

```java
    @Test
    public void parseHHMM1() throws ParseException {
        PlainTime start = PlainTime.of(7, 20);
        PlainTime end = PlainTime.of(24, 0);
        ClockInterval interval = ClockInterval.between(start, end);
        ParseLog plog = new ParseLog();
        assertThat(
            ClockInterval.parse(
                "07:20 - 24:00",
                ChronoFormatter.ofTimePattern("HH:mm ", PatternType.CLDR_24, Locale.ROOT),
                '-',
                ChronoFormatter.ofTimePattern(" HH:mm", PatternType.CLDR_24, Locale.ROOT),
                BracketPolicy.SHOW_WHEN_NON_STANDARD,
                plog),
            is(interval));
    }

    @Test
    public void parseHHMM3() throws ParseException {
        PlainTime start = PlainTime.of(7, 20);
        PlainTime end = PlainTime.of(24, 0);
        ClockInterval interval = ClockInterval.between(start, end);
        assertThat(
            ClockInterval.parse(
                "07:20 - 24:00",
                ChronoFormatter.ofTimePattern("HH:mm", PatternType.CLDR_24, Locale.US)),
            is(interval));
    }
```

The first example might also deploy the `addIgnorableWhitespace()`-approach if you have no fixed count of spaces. The last example automatically applies the interval pattern "{0} - {1}" for the locale US. However, I plan more localization support for interval patterns in the next release v3.10/4.7.
